Objects/funcobject.c

```diff
@@ -423,6 +423,30 @@ func_get_annotations(PyFunctionObject *op, void *Py_UNUSED(ignored))
         op->func_annotations = PyDict_New();
         if (op->func_annotations == NULL)
             return NULL;
+
+    }
+    if (PyTuple_CheckExact(op->func_annotations)) {
+        PyObject *anns = op->func_annotations;
+        op->func_annotations = PyDict_New();
+        if (op->func_annotations == NULL) {
+            Py_DECREF(anns);
+            return NULL;
+        }
+
+        assert(PyTuple_Size(anns) % 2 == 0);
+
+        for (Py_ssize_t i = 0; i < PyTuple_Size(anns); i += 2) {
+            int err = PyDict_SetItem(op->func_annotations,
+                                     PyTuple_GET_ITEM(anns, i),
+                                     PyTuple_GET_ITEM(anns, i + 1));
+
+            if (err < 0) {
+                Py_DECREF(op->func_annotations);
```

Comment:
`func_annotations` is already decrefed when `func_dealloc()` is called.

Reply:
Removed, thanks